android/app/src/main/assets/js/core.js

```diff
@@ -465,7 +465,10 @@ window.addEventListener('DOMContentLoaded', async () => {
     }
 
     if (reader.generalSettings.val.removeExtraParagraphSpacing) {
-      html = html.replace(/<\s*br[^>]*>/gi, '\n').replace(/\n{2,}/g, '\n\n');
+      html = html
+        .replace(/([\u200B-\u200D\uFEFF]|&nbsp;)(?<=<p>\1)/g, '')
```

Comment:
what is these kind of uncode. Do they make sense

Reply:
They do

Reply:
Could you provide some examples.

Reply:
Why you have to use positive lookbeind and numeric reference instead of simple `<p>&nbsp;</p>`

Reply:
because ``<`` are more frequent than & and zero width space. The regex will immediately fail without losing time.